pynestml/src/main/grammars/org/PyNESTML.g4

```diff
@@ -0,0 +1,329 @@
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author dplotnikov, kperun
+ */
+
+
+/**
+  Grammar representing the Simple Programming Language (SPL). It is easy to learn imperative
+  language which leans on the Python syntax.
+*/
+grammar PyNESTML;
+
+  import Tokens;
+
+  nestmlCompilationUnit : (neuron | NEWLINE )* EOF;
+  /*********************************************************************************************************************
+  * Units-Language
+  *********************************************************************************************************************/
+
+  /**
+    ASTDatatype. Represents predefined datatypes and gives a possibility to use an unit
+    datatype.
+    @attribute boolean getters for integer, real, ...
+    @attribute unitType a SI datatype
+  */
+  datatype : isInt='integer'
+           | isReal='real'
+           | isString='string'
+           | isBool='boolean'
+           | isVoid='void'
+           | unit=unitType;
+  /**
+    ASTUnitType. Represents an unit datatype. It can be a plain datatype as 'mV' or a
+    complex data type as 'mV/s'
+  */
+  unitType : leftParentheses='(' compoundUnit=unitType rightParentheses=')'
+           | base=unitType powOp='**' exponent=NUMERIC_LITERAL
+           | left=unitType (timesOp='*' | divOp='/') right=unitType
+           | unitlessLiteral=NUMERIC_LITERAL divOp='/' right=unitType
+           | unit=NAME;
+
+  /*********************************************************************************************************************
+  * Expressions-Language
+  *********************************************************************************************************************/
+  /**
+   ASTExpr, i.e., several subexpressions combined by one or more
+   operators, e.g., 10mV + V_m - (V_reset * 2)/ms ....
+   or a simple expression, e.g. 10mV.
+  */
+  expression : leftParentheses='(' term=expression rightParentheses=')'
+         | <assoc=right> left=expression powOp='**' right=expression
+         | unaryOperator term=expression
+         | left=expression (timesOp='*' | divOp='/' | moduloOp='%') right=expression
+         | left=expression (plusOp='+'  | minusOp='-') right=expression
+         | left=expression bitOperator right=expression
+         | left=expression comparisonOperator right=expression
+         | logicalNot='not' term=expression
+         | left=expression logicalOperator right=expression
+         | condition=expression '?' ifTrue=expression ':' ifNot=expression
+         | simpleExpression
+         ;
+
```

Comment:
Did you check that the priorities of the operators are comparable to the Python ones?

Reply:
Yes, I used the following reference to check if the priorities of the operators are correct * [link](https://www.tutorialspoint.com/python/operators_precedence_example.htm).  However, it states that comparison operators have a higher priority than equality operators. In my opinion, this is not relevant for us, since expression like 5 < 6 != V_m do not make any sense.